app/controllers/orders_controller.rb

```diff
@@ -31,6 +31,20 @@ def create
     end
   end
 
+  def update
+    order = Order.find(params[:order_id])
+    if order.status != "shipped"
+      order.update(status: 3)
+      order.item_orders.map do |item_order|
+        item_order.add_back_to_inventory
+      end
+      flash[:notice] = "Order-#{order.id} is now #{order.status}"
+      redirect_to "/profile"
+    else
+
+    end
```

Comment:
Why is the else block empty? should we be flashing a message when an user tries to cancel an order that has been shipped? Maybe like this:?

```suggestion
    else
      flash[:error] = "You cannot cancel a shipped order."
    end
```

Reply:
This looks great!  I didn't want to add functionality that I wasn't requested by the user story...so I left it blank because I expected there would be an additional user story that would request some functionality that exists in the else.  But I like this....it shows great user-empathy!